### PR TITLE
migrate to structured logging

### PR DESCRIPTION
**Pull request by tanjing2020**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Ref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```

NONE

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

File: pkg/scheduler/eventhandlers.go [hunk 1]
```diff
@@ -139,21 +139,21 @@ func (sched *Scheduler) deleteNodeFromCache(obj interface{}) {
...
+			klog.ErrorS(nil, "cannot convert to *v1.Node", "obj", t.Obj)
 			return
 		}
 	default:
-		klog.Errorf("cannot convert to *v1.Node: %v", t)
+		klog.ErrorS(nil, "cannot convert to *v1.Node", "type", t)
 		return
 	}
-	klog.V(3).Infof("delete event for node %q", node.Name)
+	klog.V(3).InfoS("delete event for node %q", "node", klog.KObj(node))
```
Comment (adtac) [inline pkg/scheduler/eventhandlers.go:149]:
```suggestion
	klog.V(3).InfoS("delete event for node", "node", klog.KObj(node))
```

**Review by adtac — COMMENTED** [review]
lgtm otherwise

File: pkg/scheduler/eventhandlers.go (hunk 1 above)
Comment (tanjing2020) [reply pkg/scheduler/eventhandlers.go]: Sorry, I left out some details. Thank you very much for the careful review. Done.

File: pkg/scheduler/eventhandlers.go [hunk 2]
```diff
@@ -60,7 +60,7 @@ func (sched *Scheduler) onPvcUpdate(old, new interface{}) {
 func (sched *Scheduler) onStorageClassAdd(obj interface{}) {
 	sc, ok := obj.(*storagev1.StorageClass)
 	if !ok {
-		klog.Errorf("cannot convert to *storagev1.StorageClass: %v", obj)
+		klog.ErrorS(nil, "cannot convert to *storagev1.StorageClass", "obj", obj)
```
Comment (alculquicondor) [inline pkg/scheduler/eventhandlers.go:63]: Start messages with upper case

File: pkg/scheduler/eventhandlers.go [hunk 3]
```diff
@@ -355,7 +355,7 @@ func (sched *Scheduler) skipPodUpdate(pod *v1.Pod) bool {
 	if !reflect.DeepEqual(assumedPodCopy, podCopy) {
 		return false
 	}
-	klog.V(3).Infof("Skipping pod %s/%s update", pod.Namespace, pod.Name)
+	klog.V(3).InfoS("skipping pod update", "pod", klog.KObj(pod))
```
Comment (alculquicondor) [inline pkg/scheduler/eventhandlers.go:358]: let's give more information here:
`Pod update ignored because changes won't affect scheduling`

**Review by alculquicondor — COMMENTED** [review]
/approve

**Comment by alculquicondor** [comment]
Oops, I missed that this had an lgtm already

@tanjing2020 can you send a followup PR for the suggested changes?

**Pull request merged**